admin/RemoveSpamAccounts.pl

```diff
@@ -28,46 +28,130 @@
+    "pattern|p=s"  => \$pattern,
     "dryrun|d"    => \$dry_run,
     "force|f"     => \$force,
-) or return 2;
+    "caseinsensitive|i"    => \$case_insensitive,
+    "verbose|v"     => \$verbose,
+) or usage();
 
-my $c = MusicBrainz::Server::Context->create_script_context();
-my $sql = Sql->new($c->conn);
-my $dbh = $c->dbh;
+my %allowed_columns = (
+    'name' => 1,
+    'email' => 1,
+    'website' => 1,
+    'bio' => 1,
+);
+
+sub usage {
+    warn <<EOF;
+Usage: $0 <filter> [options]
+
+FILTERS
+    -c  --column COLUMN     Specify the column used to filter accounts
+    -p  --pattern PATTERN   Specify the pattern matching column values
+
+  Allowed columns
+    name
+    email
+    website
+    bio
+
+  Patterns are case sensitive POSIX regular expressions, see
+  https://www.postgresql.org/docs/current/static/functions-matching.html#FUNCTIONS-POSIX-REGEXP
 
-my $prefix = shift;
-if (!defined $prefix || $prefix eq '') {
-    warn "no prefix given, you dolt. Refusing to do anything.\n";
+OPTIONS
+    -d, --dryrun            Perform a trial run without removing any account
+    -f, --force             Remove accounts even if they have edits/votes/OAuth tokens
+    -i, --caseinsensitive   Consider patterns as case insensitive POSIX regular expressions
+    -v, --verbose           Print filtered column additionally to id and name
+
+EXAMPLES
+    $0 --column name --dryrun --pattern '^yvanzo\$'
+        Perform a trial run of removing account of script author
+
+    $0 --column email --dryrun --pattern '\@metabrainz.org\$'
+        Perform a trial run of removing accounts of MetaBrainz team
+
+    $0 --column website --dryrun --pattern '\\<gracenote\\.com\\>'
+        Perform a trial run of removing every account linked to Gracenote
+
+    $0 --column bio --dryrun --pattern 'unicorn' --caseinsensitive
+        Perform a trial run of removing every account which dared to mention unicorn in its bio
+
+EOF
     exit(2);
+};
+
+if (!defined $column || $column eq '') {
+    warn "No filtered column given, you dolt. Refusing to do anything.\n";
+    usage();
+}
+
+if (!exists($allowed_columns{$column})) {
+    warn "Given filtered column is not allowed, you dolt. Refusing to do anything.\n";
+    usage();
+}
+
+if (!defined $pattern || $pattern eq '') {
+    warn "No matching pattern given, you dolt. Refusing to do anything.\n";
+    usage();
 }
-$prefix .= "%";
 
-my $editors = $c->sql->select_list_of_hashes("SELECT id, name FROM editor WHERE name ILIKE ?", $prefix);
+my $c = MusicBrainz::Server::Context->create_script_context();
+my $sql = Sql->new($c->conn);
+my $dbh = $c->dbh;
+
+my $regexp_operator = $case_insensitive ? '~*' : '~';
+my $editors = $c->sql->select_list_of_hashes("SELECT id, name, $column FROM editor WHERE $column $regexp_operator ?", $pattern);
 foreach my $ed (@{$editors}) {
+    my $details = "'" . $ed->{name} . "'";
+    if ($verbose && $column ne 'name') {
+        $details .=  " [${column}='" . $ed->{$column} . "']";
```

Comment:
Since you're intentionally adding surrounding `''`, I assume you want that to be a valid SQL literal, so I might suggest `$dbh->quote($ed->{$column})` instead. I suppose it doesn't matter if this is just for logging, but outputting unescaped quotes could potentially be irritating in either case. :P